crates/orderbook/src/solvable_orders.rs

```diff
@@ -208,6 +215,44 @@ fn filter_banned_user_orders(mut orders: Vec<Order>, banned_users: &HashSet<H160
     orders
 }
 
+/// Filters EIP-1271 orders whose signatures are no longer validating.
+async fn filter_invalid_signature_orders(
+    orders: Vec<Order>,
+    signature_validator: &dyn SignatureValidating,
+) -> Vec<Order> {
+    let checks = orders
+        .iter()
+        .filter_map(|order| match &order.signature {
```

Comment:
I believe this should rather be a `filter()`, no?:
```suggestion
        .filter(|order| match &order.signature {
```
Otherwise you will not be able to simply `zip()` `orders` and their `checks` back together at line 245 because `filter_map()` would drop the placeholder `None` values you need to keep the 1 to 1 matching of `orders` and `checks`.

Reply:
🤦 you are right. It worked for my manual test because I only had a single EIP-1271 order.

Will fix.

Reply:
Fixed. I also added log message to the filtering so that we can see when this happens in the logs.

We should probably add metrics to these various filters as well (but that can be a follow up PR).

Reply:
Added a unit test to verify this logic.